core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -624,7 +625,24 @@ public PlanWithProperties visitSimpleTableExecuteNode(SimpleTableExecuteNode nod
         private PlanWithProperties visitTableWriter(PlanNode node, Optional<PartitioningScheme> partitioningScheme, PlanNode source, PreferredProperties preferredProperties, TableWriterNode.WriterTarget writerTarget)
         {
             PlanWithProperties newSource = source.accept(this, preferredProperties);
+            PlanWithProperties partitionedSource = getWriterPlanWithProperties(partitioningScheme, newSource, writerTarget);
 
+            return rebaseAndDeriveProperties(node, partitionedSource);
+        }
+
+        @Override
+        public PlanWithProperties visitMergeWriter(MergeWriterNode node, PreferredProperties preferredProperties)
```

Comment:
Do you enforce somehow that joins with `merge stream` are partitioned?

Reply:
What does `merge stream` mean? I can't find a reference to that in the code base.